### ISSUE TITLE
Node ate obviously incorrect config

### ISSUE THREAD

**Issue by cthulhu-rider** [nspcc-dev/neofs-node#2276]
## Context
Node silently started with this config (attached only incorrect part):

```yaml
storage:
  shard:
    0:
      blobstor:
      - path: /srv/neofs/data0/blobovnicza
        type: blobovnicza
      - path: /srv/neofs/data0
        type: fstree
      metabase:
        path: /srv/neofs/meta/metabase0.db
      pilorama:
        path: /srv/neofs/meta/pilorama0.db
      writecache:
        path: /srv/neofs/meta/write_cache0
    default:
      blobstor:
        blobovnicza:
          depth: 1
          opened_cache_capacity: 32
          size: 4gb
          width: 8
        compress: true
        depth: 4
        perm: '0644'
        small_object_size: 150kb
      gc:
        remover_batch_size: 100
        remover_sleep_interval: 1m
      metabase:
        max_batch_size: 1000
        perm: '0644'
      pilorama:
        max_batch_delay: 10ms
        max_batch_size: 1000
        no_sync: true
      writecache:
        capacity: 64gb
        enabled: true
        max_object_size: 128mb
        memcache_capacity: 2gb
        small_object_size: 128kb
        workers_number: 30
    1:
      blobstor:
      - path: /srv/neofs/data1/blobovnicza
        type: blobovnicza
      - path: /srv/neofs/data1
        type: fstree
      metabase:
        path: /srv/neofs/meta/metabase1.db
      pilorama:
        path: /srv/neofs/meta/pilorama1.db
      writecache:
        path: /srv/neofs/meta/write_cache1
  shard_pool_size: 600
  shard_ro_error_threshold: 30
```

As we can see `storage.shard.default.blobstor` section is formed incorrectly according to [docs](https://github.com/nspcc-dev/neofs-node/blob/98e48b68514127afc291b8a8ff6b12838ed1cb5c/config/example/node.yaml#L140-L147). Also `compress` and `small_object_size` must be shard config, not blobstor.

## Expected behavior
Node run fails.

## Current behavior
Node starts silently.

## Possible improvements
* provide config validator which shares code with `neofs-node` app (maybe its command?)
* provide config defaulter which accepts source config and outputs to-be-used config (can be merged with validator)
* print the configurations used as a result in the application log

## Versions
* neofs-node v0.35.0

**Comment by roman-khimov** [comment]
@End-rey, can you try this config with the current node? This should be fixed with #2981.

**Comment by End-rey** [comment]
Such errors may appear when running with such a config:
- `unknown field: storage.shard.default.blobstor.blobovnicza`
- `unknown field: storage.shard.default.blobstor.compress`
- `unknown field: storage.shard.default.writecache.memcache_capacity`
- `unknown field: storage.shard.default.blobstor.small_object_size`

So I think the issue can be closed.
